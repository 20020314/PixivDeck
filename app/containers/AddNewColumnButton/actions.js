// @flow
import type { User } from '../../types/user'
import type { Action } from './actionTypes.js'
import { ADD_USER_ILLUST } from './constants'

export function addUserIllust(user: User): Action {
  return {
    type: ADD_USER_ILLUST,
    user,
  }
}
