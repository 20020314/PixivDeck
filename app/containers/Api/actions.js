// @flow
import type { Response } from 'services/api'

import { API_REQUEST_SUCCESS } from './actionTypes'
import type { ApiRequestSuccess } from './actionTypes'

export function apiRequestSuccess(response: Response): ApiRequestSuccess {
  return {
    type: API_REQUEST_SUCCESS,
    response,
  }
}
