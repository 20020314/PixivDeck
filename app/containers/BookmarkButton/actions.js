// @flow
import type { Restrict } from './types'

import {
  ADD_BOOKMARK_REQUEST,
  ADD_BOOKMARK_SUCCESS,
  ADD_BOOKMARK_FAILURE,
  DELETE_BOOKMARK_REQUEST,
  DELETE_BOOKMARK_SUCCESS,
  DELETE_BOOKMARK_FAILURE,
} from './actionTypes'
import type {
  AddBookmarkRequest,
  AddBookmarkSuccess,
  AddBookmarkFailure,
  DeleteBookmarkRequest,
  DeleteBookmarkSuccess,
  DeleteBookmarkFailure,
} from './actionTypes'

export function addBookmarkRequest(
  id: number,
  restrict: Restrict
): AddBookmarkRequest {
  return {
    type: ADD_BOOKMARK_REQUEST,
    id,
    restrict,
  }
}
export function addBookmarkSuccess(
  id: number,
  restrict: Restrict
): AddBookmarkSuccess {
  return {
    type: ADD_BOOKMARK_SUCCESS,
    id,
    restrict,
  }
}
export function addBookmarkFailure(
  id: number,
  error: string
): AddBookmarkFailure {
  return {
    type: ADD_BOOKMARK_FAILURE,
    id,
    error,
  }
}
export function deleteBookmarkRequest(id: number): DeleteBookmarkRequest {
  return {
    type: DELETE_BOOKMARK_REQUEST,
    id,
  }
}
export function deleteBookmarkSuccess(id: number): DeleteBookmarkSuccess {
  return {
    type: DELETE_BOOKMARK_SUCCESS,
    id,
  }
}
export function deleteBookmarkFailure(
  id: number,
  error: string
): DeleteBookmarkFailure {
  return {
    type: DELETE_BOOKMARK_FAILURE,
    id,
    error,
  }
}
