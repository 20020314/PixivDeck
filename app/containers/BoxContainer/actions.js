// @flow
import type { Action } from './actionTypes.js'
import { OPEN_PIXIV } from './constants'

export function openPixiv(id: number): Action {
  return {
    type: OPEN_PIXIV,
    id,
  }
}
