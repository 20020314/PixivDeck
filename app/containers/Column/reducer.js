// @flow
import update from 'utils/update'

export type BaseColumn = {
  illustIds: Array<number>,
  nextUrl: ?string,
}

type ActionTypes = {
  ADD_COLUMN_SUCCESS: $Subtype<string>,
  SET_NEXT_URL: $Subtype<string>,
}

export function baseReducer<
  T: BaseColumn,
  S: { [*]: T },
  AS: ActionTypes,
  A: { +type: $Subtype<string> }
>(types: AS, state: S, action: A): S {
  switch (action.type) {
    case types.SET_NEXT_URL:
      return update(state, action, { nextUrl: action.nextUrl })
    case types.ADD_COLUMN_SUCCESS:
      return update(state, action, { illustIds: [], nextUrl: null })
    default:
      return state
  }
}
