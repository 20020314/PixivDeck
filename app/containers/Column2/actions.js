// @flow
import type { Action } from './actionTypes.js'
import { ADD_COLUMN } from './constants'
import type { Body } from './reducer'

export function addColumn(id: string, body: Body): Action {
  return {
    type: ADD_COLUMN,
    id,
    body,
  }
}
