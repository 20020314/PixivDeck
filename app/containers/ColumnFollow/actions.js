// @flow
import type { Action } from './actionTypes.js'
import {
  ADD_FOLLOW_COLUMN,
  ADD_FOLLOW_COLUMN_SUCCESS,
  FETCH_FOLLOW,
  FETCH_FOLLOW_FAILRE,
  FETCH_NEW,
  FETCH_NEXT_FOLLOW,
  FETCH_NEXT_FOLLOW_FAILRE,
  SET_NEXT_URL,
  FETCH_FOLLOW_SUCCESS,
  FETCH_NEXT_FOLLOW_SUCCESS,
  FETCH_NEW_SUCCESS,
  FETCH_NEW_FAILRE,
} from './constants'
import type { Response } from 'services/api'
import type { ColumnId } from './reducer'

export function addFollowColumn(id: ColumnId): Action {
  return {
    type: ADD_FOLLOW_COLUMN,
    id,
  }
}

export function addFollowColumnSuccess(id: ColumnId): Action {
  return {
    type: ADD_FOLLOW_COLUMN_SUCCESS,
    id,
  }
}

export function fetchFollow(id: ColumnId): Action {
  return {
    type: FETCH_FOLLOW,
    id,
  }
}

export function fetchFollowFailre(id: ColumnId, error: string): Action {
  return {
    type: FETCH_FOLLOW_FAILRE,
    id,
    error,
  }
}

export function fetchNew(id: ColumnId): Action {
  return {
    type: FETCH_NEW,
    id,
  }
}

export function fetchNextFollow(id: ColumnId): Action {
  return {
    type: FETCH_NEXT_FOLLOW,
    id,
  }
}

export function fetchNextFollowFailre(id: ColumnId, error: string): Action {
  return {
    type: FETCH_NEXT_FOLLOW_FAILRE,
    id,
    error,
  }
}

export function setNextUrl(id: ColumnId, nextUrl: string): Action {
  return {
    type: SET_NEXT_URL,
    id,
    nextUrl,
  }
}

export function fetchFollowSuccess(
  id: ColumnId,
  response: Response,
  ids: Array<number>
): Action {
  return {
    type: FETCH_FOLLOW_SUCCESS,
    id,
    response,
    ids,
  }
}

export function fetchNextFollowSuccess(
  id: ColumnId,
  response: Response,
  ids: Array<number>
): Action {
  return {
    type: FETCH_NEXT_FOLLOW_SUCCESS,
    id,
    response,
    ids,
  }
}

export function fetchNewSuccess(
  id: ColumnId,
  response: Response,
  ids: Array<number>
): Action {
  return {
    type: FETCH_NEW_SUCCESS,
    id,
    response,
    ids,
  }
}

export function fetchNewFailre(id: ColumnId, error: string): Action {
  return {
    type: FETCH_NEW_FAILRE,
    id,
    error,
  }
}
