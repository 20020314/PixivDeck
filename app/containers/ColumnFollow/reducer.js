// @flow
import update from 'utils/update'
import { handleRehydrate } from 'utils/handleReydrate'
import type { Action } from './actionTypes'
import * as Actions from './constants'
import { REHYDRATE } from 'redux-persist/constants'

export type Restrict = 'public' | 'private'

export type Endpoint = '/v1/user/follows/illust'

export type ColumnId = Restrict

export type ColumnFollow = {|
  illustIds: Array<number>,
  nextUrl: ?string,
|}

export type State = $Shape<{ [ColumnId]: ColumnFollow }>

const initialState: State = {}

export default function(state: State = initialState, action: Action): State {
  switch (action.type) {
    case Actions.ADD_FOLLOW_COLUMN_SUCCESS:
      return update(state, action, { illustIds: [], nextUrl: null })
    case Actions.SET_NEXT_URL:
      return update(state, action, { nextUrl: action.nextUrl })

    case Actions.FETCH_FOLLOW_SUCCESS:
    case Actions.FETCH_NEXT_FOLLOW_SUCCESS:
      return update(state, action, { illustIds: action.ids })

    case REHYDRATE:
      return handleRehydrate(state, action, 'ColumnFollow')
    default:
      return state
  }
}
