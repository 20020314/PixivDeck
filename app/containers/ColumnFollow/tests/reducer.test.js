// @flow
import reducer from '../reducer'
import * as actions from '../actions'

const response = { result: {}, entities: {} }

test('default action', () => {
  // $FlowFixMe
  expect(reducer({}, { type: 'default action' })).toMatchSnapshot()
})

test('ADD_FOLLOW_COLUMN_SUCCESS', () => {
  expect(reducer({}, actions.addFollowColumnSuccess('public')))
  expect(
    reducer({}, actions.addFollowColumnSuccess('private'))
  ).toMatchSnapshot()
})

test('SET_NEXT_URL', () => {
  expect(
    reducer({}, actions.setNextUrl('public', 'https://nextUrl1'))
  ).toMatchSnapshot()
  expect(
    reducer({}, actions.setNextUrl('public', 'https://nextUrl/2'))
  ).toMatchSnapshot()
  expect(
    reducer({}, actions.setNextUrl('private', 'https://nextUrl/3'))
  ).toMatchSnapshot()
})

test('FETCH_FOLLOW_SUCCESS', () => {
  const first = reducer(
    {},
    actions.fetchFollowSuccess('public', response, [1, 2, 3])
  )
  expect(first).toMatchSnapshot()
  expect(
    reducer(first, actions.fetchFollowSuccess('public', response, [4, 5]))
  ).toMatchSnapshot()
})

test('FETCH_NEXT_FOLLOW_SUCCESS, FETCH_NEW_SUCCESS', () => {
  const first = reducer(
    {},
    actions.fetchNextFollowSuccess('private', response, [1, 2, 3])
  )
  expect(first).toMatchSnapshot()
  expect(
    reducer(first, actions.fetchNextFollowSuccess('private', response, [4, 5]))
  ).toMatchSnapshot()

  expect(
    reducer(first, actions.fetchNewSuccess('private', response, [8]))
  ).toMatchSnapshot()
})
