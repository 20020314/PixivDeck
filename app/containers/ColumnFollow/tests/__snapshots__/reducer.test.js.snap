// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADD_FOLLOW_COLUMN_SUCCESS 1`] = `
Object {
  "private": Object {
    "illustIds": Array [],
    "interval": 180000,
    "nextUrl": null,
  },
}
`;

exports[`FETCH_FOLLOW_SUCCESS 1`] = `
Object {
  "public": Object {
    "illustIds": Array [
      1,
      2,
      3,
    ],
  },
}
`;

exports[`FETCH_FOLLOW_SUCCESS 2`] = `
Object {
  "public": Object {
    "illustIds": Array [
      4,
      5,
    ],
  },
}
`;

exports[`FETCH_NEXT_FOLLOW_SUCCESS, FETCH_NEW_SUCCESS 1`] = `
Object {
  "private": Object {
    "illustIds": Array [
      1,
      2,
      3,
    ],
  },
}
`;

exports[`FETCH_NEXT_FOLLOW_SUCCESS, FETCH_NEW_SUCCESS 2`] = `
Object {
  "private": Object {
    "illustIds": Array [
      4,
      5,
    ],
  },
}
`;

exports[`FETCH_NEXT_FOLLOW_SUCCESS, FETCH_NEW_SUCCESS 3`] = `
Object {
  "private": Object {
    "illustIds": Array [
      8,
    ],
  },
}
`;

exports[`SET_NEXT_URL 1`] = `
Object {
  "public": Object {
    "nextUrl": "https://nextUrl1",
  },
}
`;

exports[`SET_NEXT_URL 2`] = `
Object {
  "public": Object {
    "nextUrl": "https://nextUrl/2",
  },
}
`;

exports[`SET_NEXT_URL 3`] = `
Object {
  "private": Object {
    "nextUrl": "https://nextUrl/3",
  },
}
`;

exports[`default action 1`] = `Object {}`;
